public/app/features/dashboard-scene/serialization/__snapshots__/transformSceneToSaveModelSchemaV2.test.ts.snap

```diff
@@ -0,0 +1,318 @@
+          },
+        },
+      ],
+    },
+  },
+  "links": [
+    {
+      "asDropdown": false,
+      "icon": "",
+      "includeVars": false,
+      "keepTime": false,
+      "tags": [],
+      "targetBlank": false,
+      "title": "Test Link",
+      "tooltip": "",
+      "type": "link",
+      "url": "http://test.com",
+    },
+  ],
+  "liveNow": true,
+  "preload": true,
+  "tags": [
+    "tag1",
+    "tag2",
+  ],
+  "timeSettings": {
+    "autoRefresh": "5s",
+    "autoRefreshIntervals": [
+      "5s",
+      "10s",
+      "30s",
+    ],
+    "fiscalYearStartMonth": 1,
+    "from": "now-1h",
+    "hideTimepicker": false,
+    "nowDelay": "1m",
+    "quickRanges": [],
+    "timezone": "UTC",
+    "to": "now",
+    "weekStart": "monday",
+  },
+  "title": "Test Dashboard",
+  "variables": [
+    {
+      "kind": "QueryVariable",
+      "spec": {
+        "allValue": "*",
+        "current": {
+          "text": "text1",
+          "value": "value1",
+        },
+        "datasource": {
+          "type": "prometheus",
+          "uid": "datasource1",
+        },
+        "definition": "definition1",
+        "description": "A query variable",
+        "hide": "hideLabel",
+        "includeAll": true,
+        "label": "Query Variable",
+        "multi": true,
+        "name": "queryVar",
+        "options": [],
+        "query": "query1",
+        "refresh": "onDashboardLoad",
+        "regex": "regex1",
+        "skipUrlSync": false,
+        "sort": "alphabeticalDesc",
+      },
+    },
+    {
+      "kind": "CustomVariable",
+      "spec": {
+        "allValue": "All",
+        "current": {
+          "text": "option1",
+          "value": "option1",
+        },
+        "description": "A custom variable",
+        "hide": "dontHide",
+        "includeAll": true,
+        "label": "Custom Variable",
+        "multi": true,
+        "name": "customVar",
+        "options": [
+          {
+            "selected": true,
+            "text": "option1",
+            "value": "option1",
+          },
+          {
+            "selected": false,
+            "text": "option2",
+            "value": "option2",
+          },
+        ],
+        "query": "option1, option2",
+        "skipUrlSync": false,
+      },
+    },
+    {
+      "kind": "DatasourceVariable",
+      "spec": {
+        "allValue": undefined,
+        "current": {
+          "text": "text1",
+          "value": "value1",
+        },
+        "defaultOptionEnabled": true,
+        "description": "A datasource variable",
+        "hide": "dontHide",
+        "includeAll": false,
+        "label": "Datasource Variable",
+        "multi": false,
+        "name": "datasourceVar",
+        "options": [],
+        "pluginId": "datasource1",
+        "refresh": "onDashboardLoad",
+        "regex": "regex1",
+        "skipUrlSync": false,
+      },
+    },
+    {
+      "kind": "ConstantVariable",
+      "spec": {
+        "current": {
+          "text": undefined,
```

Comment:
I'm wondering why this one is undefined when it's required as a part of https://github.com/grafana/grafana/blob/6d3a4853b3ef2ff964ef44febc6d96dd2e0a7fbf/packages/grafana-schema/src/schema/dashboard/v2alpha0/dashboard.gen.ts#L902

Reply:
The thing is we are not setting the `text` property for the ConstantVariable (and probably the interval neither). So in the transformer function i think we should either use `value` for the text of the persisted model, or see if we didn't do any special magic for these two variable types in the old architecture. But I don't think so as these don't allow text representation configuration.

Reply:
Solved. In the old arch we were using value